utils/test-utils/src/main/java/de/sovity/edc/extension/utils/Lazy.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.extension.utils;
+
+import lombok.RequiredArgsConstructor;
+
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor
+public class Lazy<T> {
```

Comment:
this is usually called a Thunk

Reply:
Never heard of that word.

This is a 
```kotlin
val x by lazy {}
```

for me.
Or the more generic concept of delegated properties.


Also it doesn't match what I found with a quick search...
https://stackoverflow.com/questions/2641489/what-is-a-thunk

So I keep `Lazy`, it will be easier when we get Kotlin around here.